### ISSUE TITLE
feature request, more supported forms of `np.round`

### ISSUE THREAD

**Issue by kenissur** [numba/numba#5110]
Hi,

I understand from [this](https://numba.pydata.org/numba-doc/dev/developer/autogen_numpy_listing.html#numpy.round_) that rounding should work in numba njit. However, this operation fails when I tried, with either np.round, np.around, np.round_, or round. Code below. Am I missing something?

```python
import numpy as np
from numba import jit


@jit(nopython=True)
def f(x):
    return np.round_(x)


f(np.array([1, 1.2, 1.8]))

```
```
Traceback (most recent call last):
  <REMOVED THIS PART, IRRELEVANT>
numba.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(<function round_ at 0x7f6e96a06cb0>) with argument(s) of type(s): (array(float64, 1d, C))
 * parameterized
In definition 0:
    All templates rejected with literals.
In definition 1:
    All templates rejected without literals.
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(<function round_ at 0x7f6e96a06cb0>)
[2] During: typing of call at <ipython-input-15-9222459bb38c> (3)`
```

```
numba.__version__
'0.47.0'
```

**Comment by esc** [comment]
@kenissur thanks for submitting this, we appreciate your input. I have updated your submission to use proper code formatting for legibility. Regarding formatting in GitHub issues, for future reference triple backticks are needed around a code block (with optional language syntax specified), for example: ` ```python <your code> ``` `. GitHub also has a basic guide to its markdown syntax available [here](https://help.github.com/articles/basic-writing-and-formatting-syntax).

**Comment by esc** [comment]
@kenissur I was able to reproduce this locally. My initial investigation suggests that there may indeed be an  issue with  `round`.

**Comment by stuartarchibald** [comment]
Looks like only one signature is supported:
http://numba.pydata.org/numba-doc/latest/developer/autogen_lower_listing.html#numpy-round
`(Array, Integer, Array)`, i.e.
```python
import numpy as np
from numba import jit


@jit(nopython=True)
def f(x):
    out = np.empty_like(x)
    return np.round(x, 2, out)


print(f(np.array([1, 1.2345, 1.8])))
```
should work.

**Comment by kenissur** [comment]
@esc Thanks!

I'll take note regarding formatting for the future :-)

**Comment by ianozsvald** [comment]
Hi, just to note that I stumbled into this with `0.53.1` whilst teaching Numba today (an attendee wanted a compiled `round` to speed-up their medical code), I don't have a PR to contribute but figured a +1 might help others looking to solve this issue. The `out=...` solution above did indeed work.

**Comment by bcbnz** [comment]
@esc  @stuartarchibald I believe this can be closed as #8234 added support for all signatures of np.round. At least, the example in the original report now works for me using the git main branch.

**Comment by esc** [comment]
> @esc @stuartarchibald I believe this can be closed as #8234 added support for all signatures of np.round. At least, the example in the original report now works for me using the git main branch.

Cool, thank you for the ping. Seems like @guilhermeleobas may know the details, can you confirm this can be closed?

**Comment by stuartarchibald** [comment]
@bcbnz @esc thanks, closing, #8234 fixed as noted.
